### ISSUE TITLE
package-name doesn't work on package designators

### ISSUE THREAD

**Issue by stassats** [drmeister/clasp#103]
(package-name "CL-USER")
(package-name :cl-user)
There are no applicable methods of COMMON-LISP:PACKAGE-NAME for receiver class COMMON-LISP:BASE-STRING/SYMBOL

**Comment by Shinmera** [comment]
Can confirm on b05884f.
```
> (package-name "CL-USER")

Condition of type: SIMPLE-PROGRAM-ERROR
There are no applicable methods of COMMON-LISP:PACKAGE-NAME for receiver class COMMON-LISP:BASE-STRING

> (package-name :cl-user)

Condition of type: SIMPLE-PROGRAM-ERROR
There are no applicable methods of COMMON-LISP:PACKAGE-NAME for receiver class COMMON-LISP:SYMBOL

```

**Comment by Shinmera** [comment]
Still there on 482d6ca5bd8ad30ff8c9b6a835989f3aec395e1a

**Comment by drmeister** [comment]
I fixed this in dea135513350b3509bc837edf4ca34bcf530158a

**Comment by Shinmera** [comment]
Fix confirmed on 9c24f9a45d0bd4bd4a80cdcac18b7cee012947fb
```
> (package-name "CL-USER")

"COMMON-LISP-USER"
> (package-name :cl-user)

"COMMON-LISP-USER"
> (package-name '#:cl-user)   

"COMMON-LISP-USER"
```
